source/overview/product.rst

```diff
@@ -40,7 +40,7 @@ Features include:
 - New features and improvements released every two months
 - Multiple languages including U.S. English, Chinese (Simplified and Traditional), Dutch, French, German, Italian, Japanese, Korean, Polish, Brazilian Portuguese, Romanian, Russian, Turkish, Spanish, and Ukrainian
 
-To get started, `download the open source Mattermost Team Edition server <https://about.mattermost.com/download>`__ under an MIT license.
+To get started, `download the open source Mattermost Team Edition server <https://docs.mattermost.com/administration/version-archive.html#mattermost-team-edition-server-archive>`__ under an MIT license.
```

Comment:
Can't comment on the earlier part of this doc -- but where it says "the open source edition is functionally equivalent...", I would recommend making a clear recommendation here as well on when to use TE and when to use EE.